docs/book/src/developer/e2e.md

```diff
@@ -198,6 +198,65 @@ long term success of the project.
 The [test E2E package] provides examples of how this can be achieved by implementing a set of reusable
 test specs for the most common Cluster API use cases.
 
+## Running the e2e tests locally
```

Comment:
We already have something similar in https://cluster-api.sigs.k8s.io/developer/testing.html#running-the-end-to-end-tests
Should we merge, at least the part about running from CLI? (While testing via IDE seems more fine in this page)

Reply:
You're right. 

I would (roughly) merge the CLI part into the existing documentation and then reference from the IDE doc here to the existing section. I think the context about Prow and how to infer the properties is helpful there and by linking it from here it should provide the necessary context.

Reply:
Wdyt does it make sense to add the Prow infos in https://cluster-api.sigs.k8s.io/developer/testing.html#running-the-end-to-end-tests ?

Reply:
Another option would be to move the entire `Running the e2e tests locally` to `Testing`. Maybe that give a cleaner separation between those docs.

Reply:
I think moving to `Testing` is logical. That could help make sure new folks are able to find it as they look into running tests.

Reply:
I'm fine with both options (move CLI only or move both)
WRT to the info about prow TBH I think we should get a dedicated page; I was chatting with Vince about the idea to open an issue for creating a page where we keep track about the kind of jobs we have (prow, github, cloudbuild), and in this page having something about how to debug test failures will be super dupe

Reply:
+1 for moving everything to Testing since that page is more about running tests and this page is more about writing new tests, with a link to running tests in this page.

Reply:
Alright sounds good to me. I'll probably rewrite it today or at least this week.